controllers/feeds/getOneFeedChat.js

```diff
@@ -37,23 +37,21 @@ const getReaction = (req, latest_reactions, constantActor) => {
   return {comments, upvotes, downvotes};
 };
 
-async function checkIsOwnPost(tokenUserId, userId) {
-  if (tokenUserId === userId) return true;
-
-  const signedUserId = UsersFunction.findSignedUserId(User, tokenUserId);
-  if (signedUserId === userId) return true;
-
-  try {
-    const anonymousUserId = await UsersFunction.findAnonymousUserId(User, tokenUserId);
-    if (anonymousUserId?.user_id === userId) return true;
-  } catch (e) {
-    return false;
-  }
-
-  return false;
-}
-
 const getOneFeedChatService = async (req, res) => {
+  let oppositeUserId = null;
+  let selfSignedUserId = null;
+  let selfAnonymousUserId = null;
+
+  if (req?.user?.is_anonymous) {
+    oppositeUserId = await UsersFunction.findSignedUserId(User, req.userId);
+    selfAnonymousUserId = req.userId;
+    selfSignedUserId = oppositeUserId;
+  } else {
+    const user = await UsersFunction.findAnonymousUserId(User, req.userId);
+    oppositeUserId = user?.id;
+    selfAnonymousUserId = user?.id;
+    selfSignedUserId = req.userId;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for determining `oppositeUserId`, `selfSignedUserId`, and `selfAnonymousUserId` seems to be incorrect. In the case where the user is not anonymous, you are assigning the same value to `oppositeUserId` and `selfAnonymousUserId`. This doesn't seem right as the opposite user and self should not be the same. Please review this logic.
```diff
  if (req?.user?.is_anonymous) {
    oppositeUserId = await UsersFunction.findSignedUserId(User, req.userId);
    selfAnonymousUserId = req.userId;
    selfSignedUserId = oppositeUserId;
  } else {
    const user = await UsersFunction.findAnonymousUserId(User, req.userId);
    oppositeUserId = user?.id;
    selfAnonymousUserId = user?.id;
    selfSignedUserId = req.userId;
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->